### PR TITLE
Functionality Redesign

### PR DESCRIPTION
**Pull request by SuperMaZingCoder**
Removed lots of code; instead, it creates a class and whenever you pull an attribute from it, the class looks at the answers file and responds accordingly. 

```py
from .answers import answers

import random


class Best:
    def __getattribute__(self, name):
        # Return random value in list
        if type(answers[name]) != list:
            return answers[name]
        else:
            return random.choice(answers[name])
```

A new file structure is also in order and can be seen here: https://github.com/SuperMaZingCoder/bestpy/tree/feature/functionality-redesign/bestpy. Refer to how each file is laid out, as well. Most of them have parts removed.

### CODE REVIEW

File: bestpy/best.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import random
+
+
+class Best:
+    def __getattribute__(self, name):
+        # Return random value in list
+        if type(answers[name]) != list:
+            return answers[name]
+        else:
+            return random.choice(answers[name])
```
Comment (vivax3794) [inline bestpy/best.py]: using isinstance and swapping around the if and else makes more sense here.

```suggestion
        if isinstance(answers[name], list):
            return random.choice(answers[name])
        else:
            return answers[name]
```

**Comment by gustavwilliam** [comment]
@SuperMaZingCoder Thanks for the PR. After the newly merged PR to add pre-commit and reformat some of the code, this can't currently be merged. Would you like to have a look at how the conflicts could be resolved?

**Comment by SuperMaZingCoder** [comment]
I'd be happy to. I'm currently not available, but I will look at it once I am \:D

**Comment by gustavwilliam** [comment]
Alright! Thanks a lot.

**Comment by SuperMaZingCoder** [comment]
@gustavwilliam All conflicts are resolved. Let me know if I need to do anything else with the PR, otherwise, merge it and let me know (here or in discord).

File: bestpy/answers.py [hunk 2]
```diff
@@ -1,34 +1,3 @@
...
-
-    @staticmethod
-    def os() -> str:
-        name = platform.system()
-
-        # Replacing "Darwin", since it might be confusing for macOS users
-        if name == "Darwin":
-            name = "macOS"
-
-        return name
```
Comment (gustavwilliam) [inline bestpy/answers.py:29]: Did this completely vanish? Any reason for that? I think the variable inputs, that change depending on the user, are quite fun to have.

**Review by gustavwilliam — COMMENTED** [review]
Overall, nice change! It fixes some of the issues we had with the previous version, such as not being able to use the package at all (a... major issue).

File: bestpy/best.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+class Best:
+    def __getattribute__(self, name):
+        # Return random value in list
+        if isinstance(answers[name], list):
+            return random.choice(answers[name])
+        # Call function
+        elif type(answers[name]).__name__ == "function":
+            return answers[name]()
+        else:
+            return answers[name]
```
Comment (gustavwilliam) [inline bestpy/best.py]:
```suggestion
    def __getattribute__(self, name):
        answer = answers[name]
        if callable(answer):
            answer = answer()
        
        if isinstance(answer, list):
            return random.choice(answer)
        return answer
```
Sometimes, the functions may return lists. We should expect the return values to act the same way that we expect the answers to act, if that makes sense. Calling it — if it's a function — could help provide the return value and let us continue parsing it as usual. This currently also looks slightly messy in my opinion.

**Review by gustavwilliam — CHANGES_REQUESTED** [review]
Overall, "I'm lovin' it". Just one comment.

**Pull request merged**